scalar/float.go

```diff
@@ -0,0 +1,220 @@
+package scalar
+
+import (
+	"math"
+	"strconv"
+
+	"github.com/apache/arrow/go/v13/arrow"
+)
+
+type Float32 struct {
+	Valid bool
+	Value float32
+}
+
+func (s *Float32) IsValid() bool {
+	return s.Valid
+}
+
+func (*Float32) DataType() arrow.DataType {
+	return arrow.PrimitiveTypes.Float32
+}
+
+func (s *Float32) String() string {
+	if !s.Valid {
+		return "(null)"
+	}
+	return strconv.FormatFloat(float64(s.Value), 'f', -1, 32)
+}
+
+func (s *Float32) Equal(rhs Scalar) bool {
+	if rhs == nil {
+		return false
+	}
+	r, ok := rhs.(*Float32)
+	if !ok {
+		return false
+	}
+	return s.Valid == r.Valid && s.Value == r.Value
+}
+
+func (s *Float32) Set(val any) error {
+	if val == nil {
+		s.Valid = false
+		return nil
+	}
+
+	switch value := val.(type) {
+	case int8:
+		s.Value = float32(value)
+	case int16:
+		if value > math.MaxInt8 {
```

Comment:
I don't really understand why this check is here - what's wrong with `int16` being given to `float32`? Seems like it should work regardless of being bigger than `math.MaxInt8`? (Same for the cases below)